### ISSUE TITLE
QuantumEspresso

### ISSUE THREAD

**Issue by Sezoir** [flang-compiler/f18-llvm-project#1457]
Enable building and execution of the QuantumEspresso application with the LLVM Flang compiler.

**Comment by Sezoir** [comment]
#### **Next steps:** I am not planning to take on this project. This is just an initial report found. Please feel free to work on this.

### Quantum ESPRESSO:
https://gitlab.com/QEF/q-e

The build system for Quantum Espresso can be built with either CMake, or using there custom configure script. Since CMake does not currently support flang-new, I'll focus on using the custom configure script in this report.

### Initial report for Quantum ESPRESSO:
At the time of writing there seems to be conflicting information between the [online documentation](https://www.quantum-espresso.org/wp-content/uploads/user_guide.pdf) (Section 2.7.1 states "Fortran and C compilers, compliant with F2003 and C89 standards") and the repository documentation (Doc/user_guide.tex which states "you need a Fortran compiler compliant with F2008 standard"). I am guessing there was a standard increase in the project as I got a compiler error when trying to compile with F2003.
- **Minimum version of Fortran required using gfortran: F2008**
- **How to compile with gfortran:**
    1. First you can download the project by running in bash:
    ```
    git clone https://gitlab.com/QEF/q-e.git
    cd q-e
    ```
    2. Now you can compile the project by running:
    ```
    ./configure FC=gfortran CC=gcc F77=gfortran
    make all
    ```
     &nbsp; &nbsp; &nbsp; Where the paths to the GNU compilers are correctly substituted in.
- **Does it compile with flang-new: No**
    Configure script does not currently recognise flang-new as a compiler, and hence defaults to gfortran.

### Versions:
- gcc version used: 9.3.0
- gfortran version used: 9.3.0
- flang-new, fir-dev branch, HEAD: fb8eb638b46eae15c89ca8fc448831a99b675c61
- clang-12: 12.0.0
- Quantum Espresso, develop branch, HEAD: 0048df7c99fc89df55d0c9405df063a9265ffbca
- Architecture for compilation: aarch64
